### PR TITLE
Add troubleshooting section for dev search docs

### PR DESCRIPTION
**Pull request by stsewd**
This is very common the first time search ins run

### CODE REVIEW

File: docs/development/search.rst [hunk 1]
```diff
@@ -65,10 +64,27 @@ with Django.
...
+Troubelshooting
+^^^^^^^^^^^^^^^
+
+If you get an error like::
+
+   RequestError(400, 'search_phase_execution_exception', 'failed to create query: ...
+
+You can fix this by deleting the page index::
+
+   inv docker.compose 'exec search curl -X DELETE search:9200/page_index'
```
Comment (humitos) [inline docs/development/search.rst:81]: I've been using `inv docker.manage 'search_index --delete'` and had the same effects. Maybe it's better to stay with django management commands.

**Comment by humitos** [comment]
> This is very common the first time search ins run

Also, if this always happens, we may want to run this in the `--init`.

**Comment by stsewd** [comment]
> Also, if this always happens, we may want to run this in the --init.

I don't think that's a good idea, it will destroy the search index every time we run it with `--init`

**Comment by humitos** [comment]
That's true. Although, the `--init` argument is receiving another meaning now. We should use it only for init.

Currently, as the workflow is right now, it's already broken. Always, the first time that you setup your development environment, you will have to use the troubleshooting section.

**Comment by ericholscher** [comment]
@stsewd @humitos we should add the search index creation to the `--init`, that should stop this issue from happening.

**Pull request merged**